src/app/lifecycle.ts

```diff
@@ -30,48 +31,60 @@ export class AppStore {
   private readonly appStore = new Map<string, FirebaseApp>();
 
   public initializeApp(options?: AppOptions, appName: string = DEFAULT_APP_NAME): App {
+    validateAppNameFormat(appName);
+
+    let autoInit = false;
     if (typeof options === 'undefined') {
+      autoInit = true
       options = loadOptionsFromEnvVar();
       options.credential = getApplicationDefault();
     }
 
-    if (typeof appName !== 'string' || appName === '') {
+    // Check if an App already exists and, if so, ensure its AppOptions match
```

Comment:
Couple of literals here too: "`App` already exists and, if so, ensure its `AppOptions`"

But maybe these comments don't get parsed for public docs; in that case, no big deal :)